### PR TITLE
Fix/enhance youtube privacy embed url

### PR DESCRIPTION
**Pull request by adriangohjw**
## Problem

- YouTube privacy-enhanced mode only rewrote `www.youtube.com`; other valid hosts (e.g. `youtube.com`) were not rewritten, bypassing privacy.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible

**Improvements**:

- Video: Rewrite all valid YouTube hosts to `www.youtube-nocookie.com` (aligned with `VALID_VIDEO_DOMAINS`). Moved logic to `utils.ts` and added unit tests.

## Tests

- As long as CI pass. This is just enhancing whats already there

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are limited to video embed URL normalization (YouTube host rewrite + path handling) and are covered by new unit tests; main impact is potentially different/empty `src` for unsupported YouTube URL shapes.
> 
> **Overview**
> Improves the `Video` component’s YouTube privacy-enhanced embedding by rewriting *all* valid YouTube hosts (per `VALID_VIDEO_DOMAINS`) to `www.youtube-nocookie.com` and normalizing `/watch?v=…` into `/embed/…`.
> 
> Refactors the privacy-enhanced embed helpers out of `Video.tsx` into `utils.ts` (including Vimeo `dnt=true` handling) and adds focused unit tests covering host/path rewrite behavior and edge cases (missing `v`, unsupported paths).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a066c164d35e49757e41b730034e48b2b68ef3d6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/components/src/templates/next/components/complex/Video/utils.ts [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+ * Rewrites a YouTube URL to the privacy-enhanced embed form (youtube-nocookie.com).
+ * Converts /watch?v=id to /embed/id. Returns undefined for unsupported path shapes.
+ */
+export const getPrivacyEnhancedYouTubeEmbedUrl = (
+  urlObject: URL,
+): string | undefined => {
+  const isYouTube = VALID_VIDEO_DOMAINS.youtube.includes(urlObject.hostname)
+  const isAlreadyPrivacyEnhanced =
+    urlObject.hostname === "www.youtube-nocookie.com" ||
+    urlObject.hostname === "youtube-nocookie.com"
```
Comment (dcshzj) [inline packages/components/src/templates/next/components/complex/Video/utils.ts]: nit: btw reviewing this code while releasing, this is not using `YOUTUBE_PRIVACY_ENHANCED_HOST` or some derivative of it, so the magic strings problem still exists
